### PR TITLE
Add shaderdb tests for centroid, noperspective interpolation

### PR DESCRIPTION
**Pull request by vettoreldaniele**
Test that `InterpLinearCentroid` is used instead of `InterpPerspCentroid`.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
Can one of the admins verify this patch?

File: llpc/test/shaderdb/extensions/OpExtInst_TestInterpolateAtCentroidNoPersp_lit.frag [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+void main()
+{
+    float f1_0 = interpolateAtCentroid(f1_1);
+
+    vec4 f4_0 = interpolateAtCentroid(f4_1);
+
+    fragColor = (f4_0.y == f1_0) ? vec4(0.0) : vec4(1.0);
+}
+// BEGIN_SHADERTEST
```
Comment (kuhar) [inline llpc/test/shaderdb/extensions/OpExtInst_TestInterpolateAtCentroidNoPersp_lit.frag:16]: Can you put the checks at the top of the file? Our test suite is not entirely consistent here, but we generally prefer to have RUN lines at the top to be able to quickly tell what the test does.

**Comment by vettoreldaniele** [comment]
V1: Move checks at the top of the file.

**Review by kuhar — DISMISSED** [review]

**Comment by kuhar** [comment]
test this please

**Comment by kuhar** [comment]
@vettoreldaniele seems like you need to rebase

**Review by kuhar — APPROVED** [review]

**Comment by kuhar** [comment]
FYI, I forgot that the rebase action does not rerun checks: https://github.com/GPUOpen-Drivers/llpc/issues/1403

**Comment by vettoreldaniele** [comment]
Force-pushed to re-run checks.

**Review by kuhar — APPROVED** [review]

**Comment by amdvlk-admin** [comment]
### Test summary for commit a8b614a07f02e5068434d432ce48fcdc56c3a1e7
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 39f8940199e60c44d4211cf8165dfd12876316fa</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev bcec463efe5260776d486a5e3da0c549bc0a75d2</li>
<li>SPVGEN: dev 580ade71d630a304805e9468ef22ad263c37d137</li>
<li>XGL: dev 04534321e450d220d404a240a047c7e1cd82f3fd</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev c6950e03a4c97beadc88a8bf5bea3dc5aa2649ee</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/225376)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 32541/56344 (57.8%)</li>
<li>Failed: 0/56344 (0.0%)</li>
<li>Not Supported: 23803/56344 (42.2%)</li><li>Warnings: 0/56344 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 32381/56344 (57.5%)</li>
<li>Failed: 0/56344 (0.0%)</li>
<li>Not Supported: 23962/56344 (42.5%)</li><li>Warnings: 1/56344 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 32717/56344 (58.1%)</li>
<li>Failed: 0/56344 (0.0%)</li>
<li>Not Supported: 23627/56344 (41.9%)</li><li>Warnings: 0/56344 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 35698/56344 (63.4%)</li>
<li>Failed: 0/56344 (0.0%)</li>
<li>Not Supported: 20646/56344 (36.6%)</li><li>Warnings: 0/56344 (0.0%)</li>
</ul></details>


</ul></details>

**Review by amdrexu — APPROVED** [review]

**Comment by amdvlk-admin** [comment]
### Test summary for commit 834719d5c796785d9333524178c8db37b397d242
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 39f8940199e60c44d4211cf8165dfd12876316fa</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev bcec463efe5260776d486a5e3da0c549bc0a75d2</li>
<li>SPVGEN: dev 580ade71d630a304805e9468ef22ad263c37d137</li>
<li>XGL: dev 04534321e450d220d404a240a047c7e1cd82f3fd</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev c6950e03a4c97beadc88a8bf5bea3dc5aa2649ee</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/225376)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 32541/56344 (57.8%)</li>
<li>Failed: 0/56344 (0.0%)</li>
<li>Not Supported: 23803/56344 (42.2%)</li><li>Warnings: 0/56344 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 32381/56344 (57.5%)</li>
<li>Failed: 0/56344 (0.0%)</li>
<li>Not Supported: 23962/56344 (42.5%)</li><li>Warnings: 1/56344 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 32717/56344 (58.1%)</li>
<li>Failed: 0/56344 (0.0%)</li>
<li>Not Supported: 23627/56344 (41.9%)</li><li>Warnings: 0/56344 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 35698/56344 (63.4%)</li>
<li>Failed: 0/56344 (0.0%)</li>
<li>Not Supported: 20646/56344 (36.6%)</li><li>Warnings: 0/56344 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request merged**